v3/lint/registration.go

```diff
@@ -118,12 +118,11 @@ func (lookup *lookupImpl[T]) register(lint *T, name string, source LintSource) e
 	if name == "" {
 		return errEmptyName
 	}
-	if existing := lookup.ByName(name); existing != nil {
-		return &errDuplicateName{name}
-	}
-
 	lookup.Lock()
 	defer lookup.Unlock()
+	if existing := lookup.lintsByName[name]; existing != nil {
```

Comment:
I went ahead and placed a write lock on the full scope of the body in order to avoid the time-of-check-time-of-use bug.